src/clr-core/modal/modal.stories.ts

```diff
@@ -0,0 +1,234 @@
+
+import { cssGroup, propertiesGroup, setStyles } from '@clr/core/internal';
+import '@clr/core/modal';
+import { action } from '@storybook/addon-actions';
+import { boolean, color as colorKnob, select, text } from '@storybook/addon-knobs';
+import { html } from 'lit-html';
+import { unsafeHTML } from 'lit-html/directives/unsafe-html';
+
+export default {
+  title: 'Components/Modal/Stories',
+  component: 'cds-modal',
+  parameters: {
+    options: { showPanel: true },
+    design: {
+      type: 'figma',
+      url: 'https://www.figma.com/file/v2mkhzKQdhECXOx8BElgdA/Clarity-UI-Library---light-2.2.0?node-id=0%3A548',
+    },
+  },
+};
+
+function htmlDecode(input: string) {
+  const e = document.createElement('div');
+  e.innerHTML = input;
+  return e.childNodes.length === 0 ? '' : e.childNodes[0].nodeValue;
+}
+
+export const API = () => {
+  const modalHeaderSlot = text('cds-modal-header', '<h3 cds-text="title">My Modal</h3>', propertiesGroup);
+  const modalContentSlot = text('cds-modal-content', '<p>This is a modal.</p>', propertiesGroup);
+  const modalFooterSlot = text(
+    'cds-modal-footer',
+    '<div cds-layout="horizontal align:right"><cds-button>Ok</cds-button></div>',
+    propertiesGroup
+  );
+
+  const closable = boolean('closable', true, propertiesGroup);
+  const size = select('size', { '(default)': 'default', sm: 'sm', lg: 'lg', xl: 'xl' }, undefined, propertiesGroup);
+
+  const backdropOpacity = text('--backdrop-opacity', undefined, cssGroup);
+  const backdropColor = colorKnob('--backdrop-color', undefined, cssGroup);
+  const backgroundColor = colorKnob('--background-color', undefined, cssGroup);
+  const boxShadowColor = colorKnob('--box-shadow-color', undefined, cssGroup);
+  const borderRadius = text('--border-radius', undefined, cssGroup);
+  const borderColor = colorKnob('--border-color', undefined, cssGroup);
+  const closeIconColor = colorKnob('--close-icon-color', undefined, cssGroup);
+  const closeIconColorHover = colorKnob('--close-icon-color-hover', undefined, cssGroup);
+  const contentBoxShadowColor = colorKnob('--content-box-shadow-color', undefined, cssGroup);
+  const width = text('--width', undefined, cssGroup);
+
+  return html`
+    <style>
+      cds-modal {
+        ${setStyles({
+          '--backdrop-opacity': backdropOpacity,
+          '--backdrop-color': backdropColor,
+          '--background-color': backgroundColor,
+          '--box-shadow-color': boxShadowColor,
+          '--border-radius': borderRadius,
+          '--border-color': borderColor,
+          '--close-icon-color': closeIconColor,
+          '--close-icon-color-hover': closeIconColorHover,
+          '--content-box-shadow-color': contentBoxShadowColor,
+          '--width': width,
+        })}
+      .sbdocs-preview > div[class^="css-"], .sbdocs-preview > div[class^="css-"] > div[class^="css-"]:first-child {
+        height: 500px;
+      }
+      body[cds-layout='no-scrolling'] {
+        overflow: auto !important;
+      }
```

Comment:
nit: I dont think the body no scroll overrides in each story demo is needed now that the focus trap can be disabled.